packages/sycamore-router/src/router.rs

```diff
@@ -370,6 +370,35 @@ pub fn navigate_replace(url: &str) {
     });
 }
 
+/// Navigates to the specified `url`. Without doing anything with the history.
+///
+/// # Panics
+/// This function will `panic!()` if a [`Router`] has not yet been created.
+pub fn navigate_no_history(url: &str) {
+    PATHNAME.with(|pathname| {
+        assert!(
+            pathname.get().is_some(),
+            "navigate_no_hist can only be used with a Router"
+        );
+
+        let pathname = pathname.get().unwrap_throw();
+        let path = url.strip_prefix(&base_pathname()).unwrap_or(url);
+        pathname.set(path.to_string());
+    });
```

Comment:
I think we should also add this line here:
```rust
window().scroll_to_with_x_and_y(0.0, 0.0);
```
Although we probably want to make this configurable in the future.